plugins/rebalance/rebalance.go

```diff
@@ -0,0 +1,109 @@
+package rebalance
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"os/exec"
+	"path"
+
+	"github.com/cespare/xxhash"
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	config "github.com/spf13/viper"
+)
+
+const (
+	// sbinDir represents bin directory
+	sbinDir = "glusterfs"
+)
+
+// RebalanceProcess type represents information about rebalance process
+type RebalanceProcess struct {
+	binarypath     string
+	args           string
+	socketfilepath string
+	pidfilepath    string
+	rebalance      RebalanceInfo
+}
+
+// Name returns the process name
+func (r *RebalanceProcess) Name() string {
+	return "rebalance"
+}
+
+// Path returns absolute path to the binary of rebalance process
+func (r *RebalanceProcess) Path() string {
+	return r.binarypath
+}
+
+// Args returns the arguments to be passed to rebalance process
+func (r *RebalanceProcess) Args() string {
+
+	shost, _, _ := net.SplitHostPort(config.GetString("clientaddress"))
+	if shost == "" {
+		shost = "localhost"
+	}
+
+	volFileID := path.Join("rebalance/", r.rebalance.Volname)
+	File := path.Join(config.GetString("logdir"), "glusterfs", r.rebalance.Volname)
+	logFile := fmt.Sprintf("%s-rebalance.log", File)
+	cmd := r.rebalance.Status
+	commitHash := r.rebalance.CommitHash
+
+	var buffer bytes.Buffer
+	buffer.WriteString(fmt.Sprintf(" -s %s", shost))
+	buffer.WriteString(fmt.Sprintf(" --volfile-id %s", volFileID))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.use-readdirp=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.lookup-unhashed=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.assert-no-child-down=yes"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.data-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.metadata-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option replicate.entry-self-heal=off"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.readdir-optimize=on"))
+	buffer.WriteString(fmt.Sprintf(" --process-name rebalance"))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.rebalance-cmd=%d", cmd))
+	buffer.WriteString(fmt.Sprintf(" --xlator-option *dht.node-uuid=%s", gdctx.MyUUID))
```

Comment:
I'm not sure what this does but it is expected to be the node-uuid of the node on which the process will be running.

Reply:
`gdctx.MyUUID` is current node's uuid where Glusterd2 is running

Reply:
Will this be run on each node of the volume and updated correctly?

Reply:
This is part of Daemon interface called from Transaction step. This will be updated correctly when run on each node.

Added a comment in Transaction step to start in all Volume nodes instead of only current node